lvsfunc/comparison.py

```diff
@@ -13,6 +17,316 @@
+
+                if key != (self.num_clips - 1):
+                    bottom_crop += overflow
+
+                self.clips[name] = clip.std.Crop(top=top_crop, bottom=bottom_crop)
+
+    def _markclips(self) -> List[vs.VideoNode]:
+        """
+        This allows for internal functions to specify a space-only string,
+        such as '   ' that avoid placing a label on the clip.
+
+        By default, `vapoursynth.core.text.Text(text='')` will not place a label,
+        but specifying multiple clips in the **clips Dict with the same '' key will not work.
+        """
+        clips: List[vs.VideoNode] = []
+        for name, clip in self.clips.items():
+            if name.strip():
+                clips.append(clip.text.Text(text=name, alignment=self.label_alignment))
+            else:
+                clips.append(clip)
+
+        return clips
+
+    @abstractmethod
+    def _compare(self) -> vs.VideoNode:
+        raise NotImplementedError
+
+    def clip(self) -> vs.VideoNode:
+        """
+        Returns the comparison as a single VideoNode for further manipulation or attribute inspection.
+
+        `comp_clip = Comparer(**kwargs, name1=clip1, name2=clip2, ...).clip()` is the intended use in encoding scripts.
+        """
+        return self._compare()
+
+    def compare(self) -> None:
+        """
+        Directly calls ``vapoursynth.VideoNode.set_output()`` on the :py:class:`lvsfunc.comparison.Comparer` generated comparison clip.
+
+        `Comparer(**kwargs, name1=clip1, name2=clip2, ...).compare()` is the intended use in encoding scripts linked with a previewer.
+        """
+        return self._compare().set_output()
+
+
+class Stack(Comparer):
+    """
+    Stacks clips horizontally or vertically.
+
+    Acts as a convenience combination function of ``vapoursynth.core.text.Text``
+    and either ``vapoursynth.core.std.StackHorizontal`` or ``vapoursynth.core.std.StackVertical`` .
+
+    :param direction:                  Enum value from :py:class:`lvsfunc.comparison.Direction`.
+    :param label_alignment:            Number from 1 to 9, corresponding to the positions of the keys on a numpad.
+                                       Determines where to place clip name using ``vapoursynth.core.text.Text`` .
+                                       Will default to top-left for horizontal comparison or top-middle for vertical comparison
+                                       if not specified (Default: None)
+    :param clips:                      Keyword argument `name=clip` pairs for comparison. `name`s will be used to label the clips.
+                                       The order the `clips` are specified will be respected in the comparison.
+                                       If stacking horizontally, all clip heights must be the same.
+                                       If stacking vertically, all clip widths must be the same.
+    """
+
+    def __init__(self, direction: Direction,
```

Comment:
If you added `direction` as a kw-only argument, you could use the same signature as Comparer.

Reply:
Can you explain this?